subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -138,6 +138,9 @@ static struct lwm2m_engine_obj *get_engine_obj(int obj_id);
 static struct lwm2m_engine_obj_inst *get_engine_obj_inst(int obj_id,
 							 int obj_inst_id);
 
+/* function pointer to indicate notification timeout to the application */
+static ext_notify_timeout_cb_t ext_notify_timeout_cb;
```

Comment:
sm_do_registration calls lwm2m_engine_start(client.ctx) in lwm2m_rd_client.c which is the first use of the lwm2m_ctx structure. If notify_timeout_cb is part of lwm2m_ctx, it should really be set in sm_do_registration, but I don't know how best to do that by the application.
 I guess we can set it later using a lwm2m_engine_set_ext_notify_timeout_cb function? Also I'm not sure if that callback can take any useful parameters, so it will be just a general notification that a timeout has occurred

Reply:
I think the solution is add notify_timeout_cb to lwm2m_ctx and set it before the application calls lwm2m_rd_client_start()

Reply:
Yes, we already do this with `tls_tag` for instance. IMO keeping thigs like this within `lwm2m_ctx` structure is more future-proof, in case we finally support connection to multiple LwM2M servers (not sure when will this happen though).